smol/cofe/space/clusterspace.py

```diff
@@ -700,6 +692,15 @@ def occupancy_from_structure(
                 if True, the occupancy string will have the index of the species
                 in the expansion structure site spaces, rather than the
                 species itself.
+        Notice: smol code is designed such that, if you use one structure
+                but two different super-cell matrices as the input, even if the two
+                matrices are symmetrically equivalent (structure matcher can match
+                prim.make_supercell), as long as the two matrices are not fully identical
+                (np.allclose), the resulting occupancy string will not be
+                the same because of different site mappings. So is the case with correlation
+                functions.
+                As a result, better always store the super-cell matrix you used when encoding
+                an occupancy!
```

Comment:
I'm confused by what exactly this notice is saying?

I dont think its necessary?

Users should really be thinking about supercell matrices at all, here (except when creating a processor). The supercell matrices are actually saved by the `StructureWrangler` as users add training data. Being able to pass them is really to avoid running StructureMatcher when users may updated ClusterSubspaces (ie by pruning them, or changing site basis). Advanced users may also find situations in which the may want to short-circuit this and pass in a supercell_matrix they obtained some other way....but those are advanced cases and as the docstring already says resposibility is on the user.

Unless I am not understanding, then I'd say we should remove this notice altogether.

Reply:
I agree if the StructureWrangler stores scmatrix perfectly for you. We can delete this notice.